src/Containers/include/Containers/ScopeTree.h

```diff
@@ -148,6 +150,27 @@ const ScopeT* ScopeTree<ScopeT>::FindFirstChild(const ScopeT& scope) const {
   return children.begin()->second->GetScope();
 }
 
+template <typename ScopeT>
+const ScopeT* ScopeTree<ScopeT>::FindFirstVisibleScopeAfterTime(uint32_t depth,
+                                                                uint64_t min_visible_time) const {
+  auto& ordered_nodes = ordered_nodes_by_depth_.at(depth);
+
+  // First we get the first node who is fully after the provided time.
+  auto node_it = ordered_nodes.upper_bound(min_visible_time);
+
+  // Previous nodes could also be partially visible if their ending is after min_visible_time.
+  auto previous_node_it = node_it;
+  while (node_it != ordered_nodes.begin() &&
+         (--previous_node_it)->second->GetScope()->end() >= min_visible_time) {
+    node_it = previous_node_it;
+  }
```

Comment:
I have a few thoughts here. 

First, let's me explain why I make the while here: Simply to make it easier to understand what we are doing here. In the current case (without overlapping timers) it's exactly the same, there should be at most one iteration of the while-loop. We had a discussion with Vicky where I explained it, and I suppose that writing a while-loop was easier to understand. I can add a comment if you want.

Second, let's jump to Overlapping timers (GPU case): There are 3 different solutions to not go through all timers one by one:
1- Your solution: _"Only checking the previous timer in the upper_bound (end_time)"_
This solution is efficient but has a case where it doesn't work properly
|------------------------------------------| T1  
   |-------------------------------------------|  T2
        |--------------------------------------------|   T3
             |-----------------------------------------------|  T4

In this case, the current solution only checking the previous one only prints T1 and T4.

2- _"The current solution using while-loop"_: 
This solution is correct. Will print the 4 timers. But it's efficient. Actually in the worst case is quadratic, which is terrible.

3-_"My future proposal (in a future PR)"_: Instead of filtering using NextPixel(end_time), making an upper_bound (NextPixel(start_time)). and maybe also filtering by end_time using NextPixel(end_time).
The solution will be efficient, but I'm not sure in the best implementation yet for GpuTrack, we might be interested in also filtering by end_time, because we might not want to see two Gpu finishing in the same pixel, but I'm not really sure and I want to experiment with it.

Reply:
I don't want to filter using start_time when we are sure that we don't have overlapping timers, so for now I was thinking on this for-loop. Do you want me to add a comment and a bug for this future proposal?

Reply:
After the discussion, coming back to an if but still we are not addressing the overlapping timers case.